File: functions/src/functions/models/game.py
from pydantic import BaseModel
from typing import List


class GuessSubmit(BaseModel):
    gameId: str
    guess: str


class GuessEntry(BaseModel):
    guess: str
    timestamp: str
    correct: bool


class GameData(BaseModel):
    gameId: str
    word: str
    drawing: str
    status: str
    score: int
    attempts: int
    maxAttempts: int
    startTime: str
    timeLimit: int
    guesses: List[GuessEntry] = []

File: web/src/types/game.ts
export interface Game {
  gameId: string;
  drawing?: string;
  status: "active" | "won" | "lost" | "timeout";
  score: number;
  attempts: number;
  maxAttempts: number;
  timeLeft: number;
  hint?: string;
  word?: string;
}

export interface GuessResponse {
  correct: boolean;
  status: string;
  score?: number;
  attempts: number;
  maxAttempts?: number;
  word?: string;
  message: string;
  hint?: string;
}

export interface GameHistoryEntry {
  guess: string;
  correct: boolean;
  timestamp: Date;
}

export interface SubmitGuessRequest {
  gameId: string;
  guess: string;
}
